smodels/tools/theoryPredictionsCombiner.py

```diff
@@ -419,16 +436,17 @@ def fun(mu):
 
         return mu_hat
 
-    def getUpperLimitOnMu(self, expected=False):
-        """get upper limit on signal strength multiplier, i.e. value for mu for
-            which CLs = 0.95
+    def _ul_preprocess(self, expected: bool = False) -> Tuple[float, float, Callable]:
+        """
+        preprocess upper limit computation
         :param expected: if True, compute expected likelihood, else observed
-        :returns: upper limit on signal strength multiplier mu
         """
-        if "UL" in self.cachedObjs[expected]:
-            return self.cachedObjs[expected]["UL"]
+        # @JACK: this needs to be fixed. Is this UL refers to UL on mu?
+        # if "UL" in self.cachedObjs[expected]:
```

Comment:
@WolfgangWaltenberger, I believe the cache check should be done outside of this preprocess function. Is this UL on mu or something else?